WordPress/gutenberg
# Media & Text: Switch default alignment to `none`

Fixes: https://github.com/WordPress/gutenberg/issues/20941

Related: 
- https://github.com/WordPress/gutenberg/pull/46971
- https://github.com/WordPress/gutenberg/issues/20941

## What?

Switches the default `align` attribute value to `none` instead of `wide` for the Media & Text block.


## Why?

Makes the block's default alignment align (pun intended) with user expectations. 

See discussion in: https://github.com/WordPress/gutenberg/issues/20941

## How?
- Updates the default value for the block's `align` attribute to `none`
- Adds a deprecation to set the `align` attribute for deprecated block's which would have previously defaulted to `wide`
- Updates the main block fixture to cover the latest version of the block
- Adds a block fixture to cover the new deprecation
- Regenerated the remaining block fixtures so their parsed and serialized results are migrated to the latest version

## Deprecation Breakdown

#### Why can we use a deprecation to change the default alignment for this block?

The prior default `wide` value for the `align` attribute resulted in the `alignwide` class being serialized within the saved block markup. When the `align` attribute is updated to default to `none`, no class is included in the markup. This makes the block's saved markup invalid and triggers the deprecation to run. The deprecation will run its `migrate` function to explicitly set the old block's `align` value to `wide`.

The `isEligible` check in the new deprecation will ensure only blocks relying on the default attribute value get migrated.

#### What about individual blocks of the previous version that explicitly set their align attribute?

If a block of the previous version relied on the default `align` value, its saved markup would be invalid thus triggering a deprecation whereas if the block set an explicit `align` value, any class saved in the markup would be correct and so the saved markup would still be valid and no migration of the `align` attribute required. 

#### Why do all prior versions of the block also need to update their align attribute?

Each deprecation is responsible for bringing its version of the block up to the latest. As a result, prior versions must keep their original attribute definitions but still migrate attributes up to the latest. So as all prior versions of the block had the default `wide` alignment, if any of those deprecated blocks relied on that default, we need to explicitly set the `align` attribute to `wide` so that that alignment is maintained.

#### What has changed with the block fixtures?

- We have a new fixture specifically to cover the new deprecation.
- The main deprecation was updated to match the `none` default alignment.
- The fixtures for prior deprecations keep their original markup and all should result in the latest version’s markup.

## Testing Instructions

TBA

Should be something like:
1. Create a post and add a Media & Text block to it
2. Save the post
3. Checkout this PR and build
4. Reload the editor
5. Confirm the block appears the same i.e. has wide alignment
6. Refresh the frontend to confirm the block is wide aligned there as well
7. Back in the editor, add a new Media & Text block
8. Confirm it was added without the wide alignment by default

## Screenshots or screencast 

TBA
